stingray/crossspectrum.py

```diff
@@ -662,7 +662,7 @@ def rebin(self, df=None, f=None, method="mean"):
                 raise AttributeError(
                     'Spectrum has no attribute named nphots2.')
 
-        bin_cs.m = np.rint(step_size * self.m)
+        bin_cs.m = np.rint(step_size[0] * self.m)
```

Comment:
How does this work? Can step_size change throughout the crossspectrum?

Reply:
Not throughout the cross spectrum, but in general, it can. Because step size encodes the number of old bins in a given new bin, it changes if the original resolution is not the same.

Reply:
But in that case, shouldn't bin_cs.m  be an array then? I mean, if step_size changes, also the number of bins averaged in each bin_cs bin will vary. We are in a similar situation as with geometrical binning